sklearn/manifold/t_sne.py

```diff
@@ -142,10 +142,12 @@ def _kl_divergence(params, P, degrees_of_freedom, n_samples, n_components,
 
     # Q is a heavy-tailed distribution: Student's t-distribution
     n = pdist(X_embedded, "sqeuclidean")
-    n += 1.
```

Comment:
t(d_ij) represents the probability density function for the t-distribution with n degrees of freedom
f(n) is useful to define, because it does not depend upon d_ij, so it makes calculating t'(d_ij) much easier!
B(1/2,n/2) represents the Beta function, although this is not necessary to include in the program since it cancels out